docusaurus/docs/engineering/knowledge-base/development-lifecycle.md

```diff
@@ -4,154 +4,239 @@ sidebar_position: 7
+
+- give an overview of the development lifecycle
+- explain key terminology
+- guide you through the step-by-step process of contributing to BobaBoard's codebase.
+
+This document also includes relevant information about git and GitHub to slowly get you familiar with more technical vocabulary you might run into. For more information on these subjects you can look at our [Git Notes](/docs/engineering/legacy-pages/git-notes.md) page.
 
 :::note
-TODO[Ms.Boba]: Explore whether the new github command line interface can make this process less cumbersome.
+We will use GitHub as our example platform since BobaBoard's code is hosted there. However, there are alternative platforms that serve similar purposes.
 :::
 
-## Development lifecycle overview
+## Overview
 
-At a high level, the git development lifecycle follows these steps:
+So, you've decided to contribute to a project. What happens next?
 
-0. **The developer "forks" the code.** This creates a separate copy of the codebase owned by the developer itself. No matter what changes the developer makes, the original codebase (called `upstream`) will remain unchanged.
-1. **The developer creates a "feature branch" on their "fork".** When the developer first forks the code, they are placed on the official "current version" of the codebase, called `master` (or `main`). Rather than modify `master` directly, the developer then creates a "feature branch" to make their edits on.
+1. **Add a copy of the project's code to your own GitHub account.**
 
-   For example, if the developer is working on adding a "read more" feature, they might create a branch called `read-more`. Creating branches allows a developer to work on many independent features at the same time.
+   > This first step involves getting your own version of the project's code to edit to your heart's content. In technical terms, this means creating a 'fork' of the project you want to contribute to.
 
-2. **The developer "commits" code changes to their "feature branch".** As the developer writes code, they will modify the code in the branch they're working on. Periodically, the developer will then "commit" the changes made. Committing creates "code checkpoints", and is helpful to split code creation in logical segment that are easier to reason about and check for correctness.
+1. **Download the code (from your GitHub account) to your computer.**
 
-   For example, when working on the `read-more` branch, the developer might create the following commits:
+   > You need to get the code onto your computer to start making changes. This way you can edit the code in the comfort of your own ~~home~~ code editor. This is done by downloading or, in technical terms, 'cloning' the code from your GitHub account.
 
-   - Commit 1: Add read more button
-   - Commit 2: Style read more button
-   - Commit 2: Display read more text when button is clicked
+1. **Create a Branch in the code to keep your changes contained.**
```

Comment:
Rather than "branch in the code" I would say a "code branch" or maybe a "branch in the repository". A code branch reads like an "if" statement to me.